Source/Core/VideoCommon/AbstractTexture.h

```diff
@@ -0,0 +1,34 @@
+// Copyright 2009 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include "Common/CommonTypes.h"
+#include "Common/MathUtil.h"
+#include "VideoCommon/TextureConfig.h"
+#include "VideoCommon/VideoCommon.h"
+
+class AbstractTexture
+{
+public:
+  explicit AbstractTexture(const TextureConfig& c)
+    : config(c)
+  {}
+
+  virtual ~AbstractTexture()
+  {}
+
+  virtual void Bind(unsigned int stage) = 0;
+  virtual bool Save(const std::string& filename, unsigned int level) = 0;
```

Comment:
Feel free to provide a default implementation here. { return false; } is fine through, as we don't depend on this feature for emulation.